### PR TITLE
[Enhancement] update aws-sdk-cpp to 1.11.267

### PR DESCRIPTION
**Pull request by silverbullet233**
## What I'm doing:

Fixes #issue

## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

**Comment by silverbullet233** [comment]
> @silverbullet233 is there a specific reason to upgrade aws sdk, how's the target version determined?

@kevincai for using s3 directory buckets

File: thirdparty/download-thirdparty.sh [hunk 1]
```diff
@@ -423,6 +423,15 @@ else
     echo "$AWS_SDK_CPP_SOURCE not patched"
 fi
 
+cd $TP_SOURCE_DIR/$AWS_SDK_CPP_SOURCE
+if [ $AWS_SDK_CPP_SOURCE = "aws-sdk-cpp-1.11.267" ]; then
```
Comment (Smith-Cruise) [inline thirdparty/download-thirdparty.sh:427]: Can you remove `aws-sdk-cpp-1.10.36` and `aws-sdk-cpp-1.9.179`, I think we don't need it anymore.

Remove relavant patch either.

If you are convenient.

**Review by Smith-Cruise — APPROVED** [review]

**Comment by kevincai** [comment]
@silverbullet233 will this be back ported to release branch?

StarOS uses aws sdk, needs an upgrade there, might be better update together with this in BE thirdparty.

File: thirdparty/download-thirdparty.sh (hunk 1 above)
Comment (silverbullet233) [reply thirdparty/download-thirdparty.sh:427]: > Can you remove `aws-sdk-cpp-1.10.36` and `aws-sdk-cpp-1.9.179`, I think we don't need it anymore.
> 
> Remove relavant patch either.
> 
> If you are convenient.

agree, I will do it

**Comment by kevincai** [comment]
included in #43887

**Pull request closed without merging**